src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -461,15 +459,20 @@ def generate_grant_sql_ddl_statements(
 
     if not extension_fn.application_roles:
         cc.warning(
-            "Skipping generation of 'GRANT USAGE ON ...' SQL statement for this object due to lack of application roles."
+            f"Skipping generation of 'GRANT USAGE ON ...' SQL statement for {extension_fn.function_type.upper()} {extension_fn.handler} due to lack of application roles."
         )
         return None
 
     grant_sql_statements = []
+    object_type = (
+        "PROCEDURE"
+        if extension_fn.function_type == ExtensionFunctionTypeEnum.PROCEDURE
+        else "FUNCTION"
+    )
```

Comment:
what was wrong with making the call? I don't see that utility getting removed, so I'm a bit confused.

Reply:
Apparently, while you do `CREATE OR REPLACE AGGREGATE FUNCTION` for create DDL, grant DDL needs to be `GRANT USAGE ON FUNCTION TO ...`, which is why we do not use the utility for grants.